src/io/perun.clj

```diff
@@ -328,25 +328,33 @@
    r renderer   RENDERER   sym  "Page renderer. Fully qualified symbol resolving to a function."
    f filterer   FILTER     code "Filter function"
    s sortby     SORTBY     code "Sort by function"
+   g groupby    GROUPBY    code "Group posts by function, keys will be used as filenames where posts (values) will be rendered"
    c comparator COMPARATOR code "Sort by comparator function"
    p page       PAGE       str  "Collection result page path"]
   (let [pods      (wrap-pool (pod/pod-pool (boot/get-env)))
         tmp       (boot/tmp-dir!)
-        options   (merge +collection-defaults+ *opts*)]
-    (cond (not (test/function? (:comparator options)))
+        options   (merge +collection-defaults+ *opts* (if-let [p (:page *opts*)]
+                                                        {:groupby (fn [_] p)}))]
+    (cond (not (fn? (:comparator options)))
               (u/fail "collection task :comparator option should be a function\n")
-          (not (test/function? (:filterer options)))
+          (not (ifn? (:filterer options)))
               (u/fail "collection task :filterer option should be a function\n")
-          (not (test/function? (:sortby options)))
+          (and (:page options) (:groupby *opts*))
+              (u/warn "using the :page option will render any :groupby option setting effectless\n")
+          (not (ifn? (:groupby options)))
+              (u/fail "collection task :groupby option should implement IFn\n")
```

Comment:
please change `should implement IFn` to `should be a function` in this case.
Or make opposite change in the other cases.

I don't have any preferences for the message itself, just care about consistency